### ISSUE TITLE
Sound issue with PipeWire

### ISSUE THREAD

**Issue by DarkArc** [flathub/net.runelite.RuneLite#16]
Hi,

There's a sound issue several people (myself included) are experiencing, a user in this [bug report](https://github.com/runelite/runelite/issues/14882) seems to have fixed it by updating to Java 18.

Could we get a version bump on the flatpak's Java runtime?

**Comment by Nightfirecat** [comment]
[The comment directly before yours on that thread](https://github.com/runelite/runelite/issues/14882#issuecomment-1135285332) says that upgrading to Java 18 does not fix it, so that simply cannot be the problem here.

**Comment by DarkArc** [comment]
As I said there, they updated the system java and ran the flatpak again (which isn't going to do anything).

**Comment by DarkArc** [comment]
Went ahead and tested this myself, can confirm updating to Java 18 fixes the issue for me.

**Comment by hankjordan** [comment]
> [The comment directly before yours on that thread](https://github.com/runelite/runelite/issues/14882#issuecomment-1135285332) says that upgrading to Java 18 does not fix it, so that simply cannot be the problem here.

The user updated his system version by mistake. After updating the Flatpak's version of Java, it resolved his issue.

**Comment by 1player** [comment]
> Went ahead and tested this myself, can confirm updating to Java 18 fixes the issue for me. (The other Fedora user also had this solved by updating to Java 18 and running it on the host system).

I have compiled the OpenJDK 18 SDK and this flatpak locally, but the audio is still broken. Did you have to do any modification to make the audio work?

**Comment by DarkArc** [comment]
> > Went ahead and tested this myself, can confirm updating to Java 18 fixes the issue for me. (The other Fedora user also had this solved by updating to Java 18 and running it on the host system).
> 
> I have compiled both the OpenJDK 18 SDK and this flatpak locally, but the audio is still broken. Did you have to do any modification to make the audio work?

You built OpenJDK (as in the flatpak itself contains OpenJDK18) or you have OpenJDK somewhere on your desktop and you built the flatpak?

There shouldn't need to be any more work than OpenJDK in the flatpak and running the game. I'm not sure the best way, but it would be a good idea to make sure OpenJDK 18 is what's actually being used in the flatpak.

For me running "bare metal" distro OpenJDK 11 and 17 (IIRC) I had the same problem, when I updated the distro OpenJDK to 18, the problem resolved.

I couldn't test this in the flatpak because there wasn't an OpenJDK 18 build. I'd assume the same principals would apply in the flatpak. There might be some "significant" build flags for the Fedora OpenJDK 18 package that are missing if you've truly got the game running on 18 but it's still not working.

**Comment by 1player** [comment]
I'm not doing anything on the host. I'm building an OpenJDK 18 flatpak (I know there isn't one, I'm building one :) but it still doesn't fix the audio issue. I've been using this PR https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk/issues/13 , updated to the latest 18 version.

I'm investigating whether we also need to upgrade the freedesktop runtime to 22.08 for audio to be fixed, because otherwise, like other commenters have said, it's not just about upgrading jdk that fixes it.

I guess I shall try recompiling OpenJDK 18 on top of the 22.08 SDK and see if that's better.

**Comment by DarkArc** [comment]
> I'm not doing anything on the host. I'm building an OpenJDK 18 flatpak (I know there isn't one, I'm building one :) but it still doesn't fix the audio issue. I've been using this PR https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk/issues/13 , updated to the latest 18 version.
> 
> But audio is still broken, so I wonder whether we also need to upgrade the freedesktop runtime to 22.08... 
> 
> I guess I shall try recompiling OpenJDK 18 on top of the 22.08 SDK and see if that's better.

It definitely seems to be going the direction of OpenJDK 18 only being part of the puzzle.

Thanks for your efforts.

**Comment by 1player** [comment]
Nope, managed to compile OpenJDK 18.0.2.1 on top of the 22.08 SDK, and audio is still broken. I'm at a loss here.

**Comment by 1player** [comment]
After two hours of pain, I have a workaround.

Create a file under `~/.var/app/net.runelite.RuneLite/config/asound.conf` with the following contents:

```
pcm.!default {
    type pipewire
    playback_node "-1"
    capture_node  "-1"
    hint {
        show on
        description "Default ALSA Output (currently PipeWire Media Server)"
    }
}

ctl.!default {
    type pipewire
}
```

Tested on my local OpenJDK 18 build, I'm curious to know if it works with the stable flathub build.

**Comment by 1player** [comment]
We can close this, it was a pipewire bug that has just been fixed: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2674

Should be hopefully available with the yet unreleased pipewire 0.3.58 version.

**Comment by 1player** [comment]
I confirm it's fixed with pipewire 0.3.58, just tested on Fedora Silverblue 36 where it's already hit the repos.
